src/vmm/src/device_manager/persist.rs

```diff
@@ -239,7 +239,7 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                                   event_manager: &mut EventManager|
          -> Result<(), Self::Error> {
             dev_manager
-                .slot_sanity_check(slot)
+                .mmio_device_info_sanity_check(slot)
```

Comment:
The only change to anything `persist` related is here. However, I don't expect this to break tests since none of the logic is changed. Nor do I expect the rest of my code to break tests since the MMIODeviceInfo struct remains unchanged.

There are a lot of warnings in the code mentioning changes that would require a "snapshot version bump," should I look into that even though I don't think any of mine require it?

Reply:
See this comment: https://github.com/firecracker-microvm/firecracker/pull/2313/files#r548548064

You can make your design work by adding a `restore_irq()` function called on the restore path just like `allocate_irq()` is called on the boot path.